src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-// write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
```

Comment:
As a recommendation: the inline arrow function added to the event listener is not very readable. It's better to create a separate function and pass it to the event listener